src/nvim/screen.c

```diff
@@ -5372,8 +5372,11 @@ void grid_puts_len(ScreenGrid *grid, char_u *text, int textlen, int row,
 
   screen_adjust_grid(&grid, &row, &col);
 
-  // safety check
-  if (grid->chars == NULL || row >= grid->Rows || col >= grid->Columns) {
+  // Safety check. The check for negative row and column is to fix issue
+  // #4102. TODO: find out why row/col could be negative.
```

Comment:
should be `vim/vim#4102`. But IIRC we normally don't mention github issues in C code, the mention in the commit message is enough.